### PR TITLE
Resolve admission plugin config files

### PR DESCRIPTION
**Pull request by liggitt**
* moves server config to testdata files
* applies defaults to test configs
* fixes admission plugin config file locations to resolve the same way all other file references in the config files are resolved

### CODE REVIEW

**Comment by liggitt** [comment]
@deads2k PTAL

**Comment by liggitt** [comment]
hmm, complaining about taking the address of a field in a struct value

**Comment by deads2k** [comment]
lgtm

**Comment by deads2k** [comment]
Any ideas on how to keep from slipping?  Make people opt-out on new fields?

**Comment by liggitt** [comment]
fixed... but wow is that ugly

**Comment by liggitt** [comment]
updated.

first commit is a straight move of the server config data into files, second commit round-trips to apply defaults to it

**Comment by liggitt** [comment]
[test]

File: pkg/cmd/server/api/types.go [hunk 1]
```diff
@@ -1233,7 +1233,7 @@ type LDAPQuery struct {
 type AdmissionPluginConfig struct {
 	// Location is the path to a configuration file that contains the plugin's
 	// configuration
-	Location string
+	Location *string
```
Comment (deads2k) [inline pkg/cmd/server/api/types.go]: so strange.

File: pkg/cmd/server/api/v1/conversions.go [hunk 2]
```diff
@@ -219,6 +219,15 @@ func addDefaultingFuncs(scheme *runtime.Scheme) error {
 				obj.ServiceAccountMethod = "prompt"
 			}
 		},
+		func(obj *AdmissionConfig) {
```
Comment (deads2k) [inline pkg/cmd/server/api/v1/conversions.go]: Why does this live in conversion and not in defaulting?

Comment (deads2k) [inline pkg/cmd/server/api/v1/conversions.go]: > Why does this live in conversion and not in defaulting?

Put another way, even if we didn't convert this, you'd still want to end up with non-nil values in the external object, right?

**Comment by deads2k** [comment]
On question about where the defaulting happens, lgtm either way.

File: pkg/cmd/server/api/v1/conversions.go (hunk 2 above)
Comment (liggitt) [inline pkg/cmd/server/api/v1/conversions.go]: This is defaulting, the defaulting functions just got added in the conversions file. I can move them all if we want to

Comment (deads2k) [inline pkg/cmd/server/api/v1/conversions.go]: > This is defaulting, the defaulting functions just got added in the conversions file. I can move them all if we want to

Ah, well that explains it.  I don't feel strongly about a move.

**Comment by liggitt** [comment]
[merge]

**Comment by liggitt** [comment]
wow, that year passed quick

**Comment by deads2k** [comment]
/lgtm

longest time in lgtm state ever?

**Pull request merged**